src/pyhf/workspace.py

```diff
@@ -0,0 +1,152 @@
+import logging
```

Comment:
@lukasheinrich Is the only reason that `Workspace` got its own file just to make CodeFactor happy? Or did you have plans to split it off already and this was a good opportunity? @kratsg had suggested moving `Workspace` and `Pdf` into a `pdf/` directory, but I guess I need some time to think on this.

Reply:
I think moving it in `workspace.py` is fine. Yeah it was just to make CF happy, but it makes sense in general too

Reply:
> but it makes sense in general too

I have to agree here.